unitTests/ui/MLFacet/MLFacetTest.ts

```diff
@@ -0,0 +1,184 @@
+
+      test.cmp.selectValue(newFacetValue);
+
+      expect(test.cmp.values.get(newFacetValue).selected).toBe(true);
+      testQueryStateModelValues();
+    });
+
+    it('allows to select multiple values', () => {
+      const selectedValues = [mockFacetValues[0].value, mockFacetValues[1].value, 'new value'];
+      expect(test.cmp.values.selectedValues.length).toBe(0);
+
+      test.cmp.selectMultipleValues(selectedValues);
+
+      expect(test.cmp.values.selectedValues).toEqual(selectedValues);
+      testQueryStateModelValues();
+    });
+
+    it('allows to deselect a value', () => {
+      mockFacetValues[2].selected = true;
+      initializeComponent();
+
+      test.cmp.deselectValue(mockFacetValues[2].value);
+
+      expect(test.cmp.values.get(mockFacetValues[2].value).selected).toBe(false);
+      testQueryStateModelValues();
+    });
+
+    it('allows to deselect multiple values', () => {
+      mockFacetValues[1].selected = true;
+      mockFacetValues[3].selected = true;
+      initializeComponent();
+
+      test.cmp.deselectMultipleValues([mockFacetValues[1].value, mockFacetValues[3].value]);
+
+      expect(test.cmp.values.selectedValues.length).toBe(0);
+      testQueryStateModelValues();
+    });
+
+    it('allows to toggle a value', () => {
+      test.cmp.toggleSelectValue(mockFacetValues[1].value);
+      expect(test.cmp.values.get(mockFacetValues[1].value).selected).toBe(true);
+      testQueryStateModelValues();
+
+      test.cmp.toggleSelectValue(mockFacetValues[1].value);
+      expect(test.cmp.values.get(mockFacetValues[1].value).selected).toBe(false);
+      testQueryStateModelValues();
+
+      test.cmp.toggleSelectValue(mockFacetValues[1].value);
+      expect(test.cmp.values.get(mockFacetValues[1].value).selected).toBe(true);
+      testQueryStateModelValues();
+    });
+
+    it('allows to trigger a new query', () => {
+      test.cmp.ensureDom();
+      test.cmp.triggerNewQuery();
+
+      expect(test.cmp.queryController.executeQuery).toHaveBeenCalled();
+    });
+
+    it('allows to reset', () => {
+      mockFacetValues[1].selected = true;
+      mockFacetValues[3].selected = true;
+      initializeComponent();
+      test.cmp.ensureDom();
+      expect(test.cmp.values.selectedValues.length).toBe(2);
+
+      test.cmp.reset();
+
+      expect($$(test.cmp.element).hasClass('coveo-active')).toBe(false);
+      expect(test.cmp.values.selectedValues.length).toBe(0);
+      testQueryStateModelValues();
+    });
+
+    it('should have a default title', () => {
+      test.cmp.ensureDom();
+
+      expect($$(test.cmp.element).find('.coveo-facet-header-title').innerHTML).toBe('No title');
+    });
+
+    it('title option should set the title', () => {
+      options.title = 'a title';
+      initializeComponent();
+      test.cmp.ensureDom();
+
+      expect($$(test.cmp.element).find('.coveo-facet-header-title').innerHTML).toBe(options.title);
+    });
+
+    describe('with a live query state model', () => {
```

Comment:
Is this `describe` block needed considering `initializeComponent` always uses a live query state model?

Reply:
right, I should remove it now that I always use the advanced mock with the qsm